File: src/test/java/org/goafabric/llm/logic/PersonLogicTest.java
package org.goafabric.llm.logic;

import org.goafabric.llm.dbagent.logic.PersonLogic;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
class PersonLogicTest {
    @Autowired
    PersonLogic personLogic;

    @Test
    void findAll() {
        assertThat(personLogic.findAll()).hasSize(4);
    }


    @Test
    void findByFirstName() {
        var person = personLogic.findByFirstName("homer");
        System.out.println(person);
        assertThat(person).hasSize(1);
        assertThat(person.getFirst().getAddress()).hasSize(1);
        assertThat(person.getFirst().getAllergy()).hasSize(1);
    }

    @Test
    void findByLastName() {
        var person = personLogic.findByLastName("sampson");

        System.out.println(person);
        assertThat(person).hasSize(2);
        assertThat(person.getFirst().getAddress()).hasSize(1);
        assertThat(person.getFirst().getAllergy()).hasSize(1);
    }

    @Test
    void findByCity() {
        var person = personLogic.findByCity("springfield");

        System.out.println(person);
        assertThat(person).hasSize(3);
        assertThat(person.getFirst().getAddress()).hasSize(1);
        assertThat(person.getFirst().getAllergy()).hasSize(1);

    }
}

File: src/test/kotlin/org/goafabric/llm/logic/PersonLogicTest.kt
package org.goafabric.llm.logic

import org.assertj.core.api.Assertions
import org.goafabric.llm.tool.logic.PersonLogic
import org.goafabric.llm.tool.persistence.entity.Address
import org.goafabric.llm.tool.persistence.entity.Allergy
import org.goafabric.llm.tool.persistence.entity.Person
import org.goafabric.llm.tool.persistence.entity.Person.address
import org.goafabric.llm.tool.persistence.entity.Person.allergy
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
internal class PersonLogicTest {
    @JvmField
    @Autowired
    var personLogic: PersonLogic? = null

    @Test
    fun findAll() {
        Assertions.assertThat<Person>(personLogic!!.findAll()).hasSize(4)
    }


    @Test
    fun findByFirstName() {
        val person: MutableList<Person?> = personLogic!!.findByFirstName("homer")
        println(person)
        Assertions.assertThat<Person>(person).hasSize(1)
        Assertions.assertThat<Address>(person.getFirst().address).hasSize(1)
        Assertions.assertThat<Allergy>(person.getFirst().allergy).hasSize(1)
    }

    @Test
    fun findByLastName() {
        val person: MutableList<Person?> = personLogic!!.findByLastName("sampson")

        println(person)
        Assertions.assertThat<Person>(person).hasSize(2)
        Assertions.assertThat<Address>(person.getFirst().address).hasSize(1)
        Assertions.assertThat<Allergy>(person.getFirst().allergy).hasSize(1)
    }

    @Test
    fun findByCity() {
        val person: MutableList<Person?> = personLogic!!.findByCity("springfield")

        println(person)
        Assertions.assertThat<Person>(person).hasSize(3)
        Assertions.assertThat<Address>(person.getFirst().address).hasSize(1)
        Assertions.assertThat<Allergy>(person.getFirst().allergy).hasSize(1)
    }
}